manual/manual/cmds/intf-c.etex

```diff
@@ -2176,15 +2176,19 @@ CAMLprim stub_gethostbyname(value vname)
   CAMLparam1 (vname);
   CAMLlocal1 (vres);
   struct hostent * h;
+  char * name;
 
   /* Copy the string argument to a C string, allocated outside the
      OCaml heap. */
-  name = stat_alloc(caml_string_length(vname) + 1);
-  name = caml_strdup (String_val(vname));
+  name = caml_stat_alloc(caml_string_length(vname) + 1);
+  name = caml_strdup(String_val(vname));
   /* Release the OCaml run-time system */
   caml_release_runtime_system();
   /* Resolve the name */
   h = gethostbyname(name);
+  /* Free the copy of the string, which we might as well do before
+     acquiring the runtime system to benefit from parallelism. */
```

Comment:
I don't understand the comment. You say that you are "freeing now to benefit from parallelism", but isn't freeing now just the natural thing to do anyway? Is this really parallelism-related?

(As a functional programming I would be tempted to put a block scope between the `release .. acquire` calls and define `char * name` in there, but that is not idiomatic C style so let's not.)